client/local/model/resource_spec.go

```diff
@@ -2,32 +2,55 @@ package model
 
 import (
 	"fmt"
+	"time"
 
 	"google.golang.org/protobuf/types/known/structpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
 
 	pb "github.com/goto/optimus/protos/gotocompany/optimus/core/v1beta1"
 )
 
+type Deprecation struct {
+	Reason           string `yaml:"reason"`
+	Date             string `yaml:"date"`
+	ReplacementTable string `yaml:"replacement_table"`
+}
+
 type ResourceSpec struct {
-	Version int                    `yaml:"version"`
-	Name    string                 `yaml:"name"`
-	Type    string                 `yaml:"type"`
-	Labels  map[string]string      `yaml:"labels"`
-	Spec    map[string]interface{} `yaml:"spec"`
-	Path    string                 `yaml:"-"`
+	Version     int                    `yaml:"version"`
+	Name        string                 `yaml:"name"`
+	Type        string                 `yaml:"type"`
+	Labels      map[string]string      `yaml:"labels"`
+	Spec        map[string]interface{} `yaml:"spec"`
+	Path        string                 `yaml:"-"`
+	Deprecation *Deprecation           `yaml:"deprecation"`
 }
 
 func (r ResourceSpec) ToProto() (*pb.ResourceSpecification, error) {
 	specPb, err := structpb.NewStruct(r.Spec)
 	if err != nil {
 		return nil, fmt.Errorf("error constructing spec pb: %w", err)
 	}
+
+	var deprecation *pb.Deprecation
+	if r.Deprecation != nil && len(r.Deprecation.Date) > 0 {
+		deprecationDate, err := time.Parse(time.DateOnly, r.Deprecation.Date)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing deprivation Date, for resource: %s, expected_format: [YYYY-MM-DD], got: [%s], err: %w", r.Name, r.Deprecation.Date, err)
```

Comment:
there was some typo in the message
```suggestion
			return nil, fmt.Errorf("error parsing deprecation Date, for resource: %s, expected_format: [YYYY-MM-DD], got: [%s], err: %w", r.Name, r.Deprecation.Date, err)
```